### PR TITLE
SAK-43579 - Rubrics: Ability to search rubrics and shared rubrics

### PR DESCRIPTION
**Pull request by mpellicer**
https://jira.sakaiproject.org/browse/SAK-43579

### CODE REVIEW

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- rubrics/tool/src/main/frontend/webcomponents/js/sakai-rubrics-manager.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=5395221&bid=17685970)

File: rubrics/tool/src/main/frontend/webcomponents/js/sakai-rubrics-manager.js [hunk 1]
```diff
@@ -104,6 +111,27 @@ class SakaiRubricsManager extends RubricsElement {
...
+    var searchInput = document.getElementById('rubrics-search-bar');
+    var searchInputValue = searchInput.value.toLowerCase();
+
+    $([rubricsList, sharedRubricsList]).find('.rubric-item').each(function () {
+      var rubricItem = $(this);
+      rubricItem.removeClass('hidden');
+      var rubricTitle = $(this).find('.rubric-name').text();
+      var rubricAuthor = $(this).find('sakai-rubric-creator-name').text();
+      var rubricSite = $(this).find('sakai-rubric-site-title').text();
+      if (!rubricAuthor.toLowerCase().includes(searchInputValue) &&
```
Comment (adrianfish) [inline rubrics/tool/src/main/frontend/webcomponents/js/sakai-rubrics-manager.js:127]: You don't need to wrap the element in a jquery object repeatedly. You're already creating the rubricsList.

File: rubrics/tool/src/main/frontend/webcomponents/js/sakai-rubrics-manager.js [hunk 2]
```diff
@@ -104,6 +111,27 @@ class SakaiRubricsManager extends RubricsElement {
       icon.removeClass("fa-chevron-down").addClass("fa-chevron-right");
     }
   }
+
+  filterRubrics() {
```
Comment (profmikegreene) [inline rubrics/tool/src/main/frontend/webcomponents/js/sakai-rubrics-manager.js:115]: Would array.filter be the ideal way to remove jquery from this? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- rubrics/tool/src/main/frontend/webcomponents/js/sakai-rubrics-manager.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=5395221&bid=17685970)

**Pull request merged**